File: easy/cs/FuncCall.cs
using common;
using iface;
using cmd_exec;

public class FuncCall {
    public static void RunByChild(string cmd) {
        VulnChild child = new VulnChild();
        child.Exec(cmd);
    }

    public static void RunByInterface(string cmd) {
        IVulnRunner runner = new InterfaceImpl();
        runner.Run(cmd);
    }
}

File: easy/scala/FuncCall.scala
import common.VulnChild
import cmd_exec.InterfaceImpl
import iface.VulnRunner

object FuncCall {
  def runByChild(cmd: String): Unit = {
    val child = new VulnChild()
    child.exec(cmd)
  }

  def runByInterface(cmd: String): Unit = {
    val runner: VulnRunner = new InterfaceImpl()
    runner.run(cmd)
  }
}
